src/alert/__tests__/runtime-content.test.tsx

```diff
@@ -0,0 +1,305 @@
+import awsuiPlugins from '../../../lib/components/internal/plugins';
+import { awsuiPluginsInternal } from '../../../lib/components/internal/plugins/api';
+import { AlertFlashContentConfig } from '../../../lib/components/internal/plugins/controllers/alert-flash-content';
+import { AlertWrapper } from '../../../lib/components/test-utils/dom';
+
+import stylesCss from '../../../lib/components/alert/styles.css.js';
+
+const pause = (timeout: number) => new Promise(resolve => setTimeout(resolve, timeout));
+
+const expectAlertContent = (
+  wrapper: AlertWrapper,
+  {
+    header,
+    headerReplaced,
+    content,
+    contentReplaced,
+  }: {
+    header?: string | false;
+    headerReplaced?: boolean;
+    content?: string | false;
+    contentReplaced?: boolean;
+  }
+) => {
+  if (header) {
+    if (headerReplaced) {
+      if (wrapper.findHeader()) {
+        expect(wrapper.findHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+      }
+      expect(wrapper.findReplacementHeader()?.getElement().textContent).toBe(header);
+    } else {
+      expect(wrapper.findReplacementHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findHeader()?.getElement().textContent).toBe(header);
+    }
+  } else if (header === false) {
+    if (wrapper.findHeader()) {
+      expect(wrapper.findHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+    }
+    expect(wrapper.findReplacementHeader()?.getElement()).toHaveClass(stylesCss.hidden);
+  }
+  if (content) {
+    if (contentReplaced) {
+      expect(wrapper.findContent().getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findReplacementContent().getElement().textContent).toBe(content);
+    } else {
+      expect(wrapper.findReplacementContent().getElement()).toHaveClass(stylesCss.hidden);
+      expect(wrapper.findContent().getElement().textContent).toBe(content);
+    }
+  } else if (content === false) {
+    expect(wrapper.findContent().getElement()).toHaveClass(stylesCss.hidden);
+    expect(wrapper.findReplacementContent().getElement()).toHaveClass(stylesCss.hidden);
+  }
+};
+
+const defaultContent: AlertFlashContentConfig = {
+  id: 'test-content',
+  runReplacer(context, registerReplacement) {
+    registerReplacement('header', container => {
+      container.append('New header');
+    });
+    registerReplacement('content', container => {
+      container.append('New content');
+    });
+    return {
+      update: () => {},
+      unmount: () => {},
+    };
+  },
+};
+
+function delay(advanceBy = 1) {
+  const promise = act(() => new Promise(resolve => setTimeout(resolve)));
+  jest.advanceTimersByTime(advanceBy);
+  return promise;
+}
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+afterEach(() => {
+  awsuiPluginsInternal.alertContent.clearRegisteredReplacer();
+  jest.useRealTimers();
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+});
+
+test('renders runtime content initially', async () => {
```

Comment:
Should we say "replaced" or "replacement" instead of "runtime"? It wasn't immediately clear if runtime means original (rendered with React at runtime) or replacement.